### ISSUE TITLE
Joy-1, -2, and -3 duplicate valid joysticks

### ISSUE THREAD

**Issue by z64555** [scp-fs2open/fs2open.github.com#4792]
This happens when only 1 or 2 joysticks are defined within the .cfg.  Essentially the joystick with device_id of 0 is assumed to be Joy-1, -2, and -3 should any of them not be defined.

This is likely due to a bug within `joy-sdl.cpp::isPlayerJoystick()`.  There is no signal currently to `isPlayerJoystick()` to determine an undefined Joystick entry, possibly because `os_config_read_int()`.

**Comment by notimaginative** [comment]
This looks to just be an issue with the order of the loops. I don't know if how it is was intentional or just an oversight, but the following change fixes it for me:

```
--- a/code/io/joy-sdl.cpp
+++ b/code/io/joy-sdl.cpp
@@ -1002,9 +1002,9 @@ namespace joystick
        else
        {
            // Fall back to the legacy config values
-           for (short i = CID_JOY0; i < CID_JOY_MAX; ++i)
+           for (auto& stick : joysticks)
            {
-               for (auto& stick : joysticks)
+               for (short i = CID_JOY0; i < CID_JOY_MAX; ++i)
                {
                    if (isPlayerJoystick(stick.get(), i))
                    {
```

**Comment by Goober5000** [comment]
Is this a fix that could/should be included in 23.0?

**Comment by notimaginative** [comment]
As far as I can tell that change fixes the problem and doesn't create any others. I'll submit an actual PR for it in just a bit and we can sort out whether it's the right fix then.
